packages/nextjs/components/scaffold-eth/Contract/ReadOnlyFunctionForm.tsx

```diff
@@ -44,7 +42,10 @@ export const ReadOnlyFunctionForm = ({ functionFragment, contractAddress }: TRea
     return (
       <InputUI
         key={key}
-        setForm={setForm}
+        setForm={updatedFormValue => {
+          setResult(null);
```

Comment:
nitpick: add `if (result !== null)` for this `setResult` since it can add renders.
And I think it should be the default value, `undefined`, or set default value `null` to `useState`

Reply:
>  add if (result !== null) for this setResult since it can add renders.

I think when you have multiple state changes (in the same event loop cycle i.e. same handler), React batches them so it doesn't add any extra renders.

Reply:
Yes, especially in such a small component. But it's not a 100% stable thing 🤷‍♂️ . Not critical at all :)

Reply:
> I think it should be the default value, undefined, or set default value null to useState

Since we were setting the value that's why setted it to null but fixed it at [54e2b64](https://github.com/scaffold-eth/se-2/pull/177/commits/54e2b64d78c16772d5bd7ea814883a2afe9adbcd) thanks 🙌

Reply:
Thanks! Added a small fix (default state was already undefined) https://github.com/scaffold-eth/se-2/pull/177/commits/09e0af11a3098ab08bc3b2adc93fe51213e220f1